api/v1alpha1/bundle_types.go

```diff
@@ -18,48 +18,73 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
-	runtime "k8s.io/apimachinery/pkg/runtime"
 )
 
-// Source locates content to stage.
-type Source string
+type ProvisionerID string
+type BundleConditionType string
+
+const (
+	TypeUnpacked = "Unpacked"
+
+	ReasonUnpackPending    = "UnpackPending"
+	ReasonUnpacking        = "Unpacking"
+	ReasonUnpackSuccessful = "UnpackSuccessful"
+	ReasonUnpackFailed     = "UnpackFailed"
+
+	PhasePending   = "Pending"
+	PhaseUnpacking = "Unpacking"
+	PhaseFailing   = "Failing"
+	PhaseUnpacked  = "Unpacked"
+)
 
 // BundleSpec defines the desired state of Bundle
 type BundleSpec struct {
-	// Class specifies the name of the provisioner that should manage the Bundle.
-	// +optional
-	Class string `json:"class,omitempty"`
+	// ProvisionerClassName sets the name of the provisioner that should reconcile this BundleInstance.
+	ProvisionerClassName string `json:"provisionerClassName"`
 
-	// Source locates all remote content to be staged by the Bundle.
-	Source Source `json:"source"`
-}
+	// Image is the bundle image that backs the content of this bundle.
+	Image string `json:"image"`
 
-// Content surfaces staged content for clients to access.
-type Content struct {
-	runtime.RawExtension `json:"-"`
+	// ImagePullSecrets is a list of pull secrets to have available to
+	// pull the referenced image.
+	ImagePullSecrets []ImagePullSecret `json:"imagePullSecrets,omitempty"`
 }
 
-type ProvisionerID string
+type ImagePullSecret struct {
+	Namespace string `json:"namespace"`
+	Name      string `json:"name"`
+}
 
 // BundleStatus defines the observed state of Bundle
 type BundleStatus struct {
-	// Provisioner is the ID of the provisioner managing this Bundle.
-	// +optional
-	Provisioner ProvisionerID `json:"provisioner,omitempty"`
+	Info               *BundleInfo        `json:"info,omitempty"`
+	Phase              string             `json:"phase,omitempty"`
+	Digest             string             `json:"digest,omitempty"`
+	ObservedGeneration int64              `json:"observedGeneration,omitempty"`
+	Conditions         []metav1.Condition `json:"conditions,omitempty"`
+}
 
-	// Contents provides access to all staged bundle content.
-	// +optional
-	Contents []Content `json:"contents,omitempty"`
+type BundleInfo struct {
+	Package string         `json:"package"`
+	Name    string         `json:"name"`
+	Version string         `json:"version"`
+	Objects []BundleObject `json:"objects,omitempty"`
+}
 
-	Conditions []metav1.Condition `json:"conditions,omitempty"`
+type BundleObject struct {
+	Group     string `json:"group"`
+	Version   string `json:"version"`
+	Kind      string `json:"kind"`
+	Name      string `json:"name"`
+	Namespace string `json:"namespace"`
```

Comment:
Another note: we had talked about whether this implementation detail also works with Bundles that contain contents that aren't GVK-based, e.g. a Bundle that contains OPA rules, and whether those use cases make sense to support. I think this is fine to keep in the short term, and we can re-evaluate over time as we continue to iterate, and make any breaking changes before GA.